cpp/include/cudf/detail/indexalator.cuh

```diff
@@ -93,7 +93,7 @@ struct input_indexalator : base_normalator<input_indexalator, cudf::size_type> {
    */
   __device__ inline cudf::size_type operator[](size_type idx) const
   {
-    void const* tp = p_ + (idx * this->width_);
+    void const* tp = p_ + (static_cast<int64_t>(idx) * this->width_);
```

Comment:
Should this be `std::ptrdiff_t`?

Reply:
I suppose. Technically if `size_type` increased to `int64_t` then `ptrdiff_t` would be more correct.